scripts/scene/RoomBuilder.gd

```diff
@@ -40,25 +41,19 @@ func _init(gui: GUI, station: Station, base_tile_map: TileMap, build_tile_map: T
 	self.rooms = rooms
 	self.room_types = room_types
 
-func start_editing() -> void:
-	selected_room_type = get_room_type_by_id(selected_room_type_id)	
-	var new_room = Room.new()
-	new_room.id = generate_unique_room_id()
-	new_room.roomTypeId = selected_room_type.id
-	rooms.append(new_room)
-
 func stop_editing() -> void:
 	selected_room_type_id = 0 # Deselect
 	build_tile_map.clear_layer(drafting_layer)
 	action_completed.emit(Action.BACK)
 
 # --- Input functions ---
+
 func selecting_tile(event: InputEventMouseButton, offset: Vector2, zoom: Vector2, selected_room_type_id: int) -> void:
 	var coords = base_tile_map.local_to_map((event.position / zoom) + offset)
 	if !any_invalid:
 		self.selected_room_type_id = selected_room_type_id
+		self.selected_room_type = get_room_type_by_id(selected_room_type_id)
```

Comment:
I wanted to do this in a `setting_roomtype` function, but it isn't triggered by an input. Really, we don't need self.selected_room_type_id if we can pass the whole selected_room_type object to RoomBuilder. I think we tried before, but only managed with the ID by binding it to the button (more complex data wouldn't pass). Perhaps we should reinvestigate.